hooks/command

```diff
@@ -136,6 +136,14 @@ do
   if [[ -n "$pod_name" ]]; then
     break
   else
+    set +e
+    jobstatus=$(kubectl get job "$job_name" -o 'jsonpath={.status.conditions[].type}')
+    set -e
+    if [[ "$jobstatus" == "Failed" ]]; then
+      echo "Warning: job failed to create pod"
+      exit 42
+    fi
```

Comment:
Let's only enter this block if the response from get pod is a pod not found, since that would be more accurate (if it fails for another reason then the pod might actually exist in which case we want to retry even if the job failed)

Reply:
If the job failed is it actually recoverable regardless of the pod status? My assumption is no?

Reply:
right, if we know the job failed then there's no need to retry. But in this case we don't need to check if the job failed.
If retrieving a pod fails for another reason e.g timeout/ratelimit, it clearer to retry that request rather than make another request that will likely fail for the same reason.
The original intent was to check if the job failed, when we don't have a pod. If we do have a pod for a failed job then we don't want to exit because we still need the pod logs etc